File: common/src/main/java/net/impactdev/gts/common/config/types/time/TimeKey.java
package net.impactdev.gts.common.config.types.time;

import net.impactdev.impactor.api.configuration.ConfigurationAdapter;
import net.impactdev.impactor.api.configuration.keys.BaseConfigKey;
import net.impactdev.impactor.api.utilities.Time;

public class TimeKey extends BaseConfigKey<Time> {

    private final String key;
    private final String def;

    public TimeKey(String key, String def) {
        this.key = key;
        this.def = def;
    }

    @Override
    public Time get(ConfigurationAdapter adapter) {
        return new Time(adapter.getString(this.key, this.def));
    }

}

File: common/src/main/kotlin/net/impactdev/gts/common/config/types/time/TimeKey.kt
package net.impactdev.gts.common.config.types.time

import net.impactdev.impactor.api.configuration.ConfigurationAdapter
import net.impactdev.impactor.api.configuration.keys.BaseConfigKey
import net.impactdev.impactor.api.utilities.Time

class TimeKey(private val key: String, private val def: String) : BaseConfigKey<Time>() {
    override fun get(adapter: ConfigurationAdapter): Time {
        return Time(adapter.getString(key, def))
    }
}